engine/src/main/java/org/hibernate/search/engine/search/reference/SearchValueFieldReference.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.reference;
+
+import org.hibernate.search.engine.search.common.ValueConvert;
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ */
+@Incubating
+public interface SearchValueFieldReference<T, P> extends ProjectionTypedFieldReference<P>, TypedFieldReference<T> {
+
+	default ValueConvert valueConvert() {
+		return ValueConvert.YES;
+	}
+
+}
```

Comment:
This `SearchValueFieldReference` takes `TypedFieldReference` to work in the predicates etc, and `ProjectionTypedFieldReference` to work in projections. This way, we can use the same reference in all places and have them use different types depending on the configured converter.